lib/logstash/inputs/jms.rb

```diff
@@ -0,0 +1,300 @@
+				:jndi_context => @jndi_context}
+
+		elsif @factory
+			@jms_config = {
+				:require_jars => @require_jars,
+				:factory => @factory,
+				:username => @username,
+				:password => @password,
+				:broker_url => @broker_url,
+				:url => @broker_url # "broker_url" is named "url" with Oracle AQ
+				}
+		end
+
+		@logger.debug("JMS Config being used", :context => @jms_config)
+
+	end # def register
+
+
+	private
+	def queue_event(msg, output_queue)
+		begin
+				event = LogStash::Event.new
+
+				# Here, we can use the JMS Enqueue timestamp as the @timestamp
+				if @use_jms_timestamp and msg.jms_timestamp
+					event.timestamp = ::Time.at(msg.jms_timestamp/1000)
+				end
+
+				if @include_header
+				 #event.append(msg.attributes)
+					msg.attributes.each do |field, value|
+						event[field.to_s] = value
+					end
+				end
+
+				if @include_properties
+				 #event.append(msg.properties)
+					msg.properties.each do |field, value|
+						event[field.to_s] = value
+					end
+				end
+
+				if @include_body
+					if msg.java_kind_of?(JMS::MapMessage)
+					 #event.append(msg.data)
+						msg.data.each do |field, value|
+							event[field.to_s] = value # TODO(claveau): needs codec.decode or converter.convert ?
+						end
+
+					elsif msg.java_kind_of?(JMS::TextMessage)
+						@codec.decode(msg.to_s) do |event_message|
+							# Copy out the header data into the message.
+							event.to_hash.each do |k,v|
+								event_message[k] = v
+							end
+							# Now lets overwrite the event.
+							event = event_message
+						end
+					else
+						@logger.error( "Unknown data type #{msg.data.class.to_s} in Message" )
+					end
+				end
+
+				decorate(event)
+				output_queue << event
+
+		rescue => e # parse or event creation error
+			@logger.error("Failed to create event", :message => msg, :exception => e,
+										:backtrace => e.backtrace);
+		end
+	end
+
+
+
+	# Consume all available messages on the queue
+	# sleeps some time, then consume again
+	private
+	def run_consumer(output_queue)
+		JMS::Connection.session(@jms_config) do |session|
+			while(true)
+				if (@pub_sub)
+					session.consume(:topic_name => @descriptor, :timeout=>@timeout, :selector => @selector) do |message|
+						queue_event message, output_queue
+					end
+				else
+					session.consume(:queue_name => @descriptor, :timeout=>@timeout, :selector => @selector) do |message|
+						queue_event message, output_queue
+					end
+				end
+				sleep @interval
+			end
+		end
+	rescue LogStash::ShutdownSignal
+		# Do nothing, let us quit.
+	rescue => e
+		@logger.warn("JMS Consumer died", :exception => e, :backtrace => e.backtrace)
+		sleep(10)
+		retry
+	end # def run
+
+
+
+
+	# Consume all available messages on the queue through a listener
+	private
+	def run_thread(output_queue)
+		connection = JMS::Connection.new(@jms_config)
+		connection.on_exception do |jms_exception|
+			@logger.warn("JMS Exception has occurred: #{jms_exception}")
+		end
+		if (@pub_sub)
```

Comment:
So the if statements are there since it expects queue_name or topic_name instead of descriptor. Ruby is not the language i use for my day job so if there is a way to do this that i am missing i am all ears since i don't like these if statements either.